src/status_im/contexts/chat/group_create/events.cljs

```diff
@@ -0,0 +1,40 @@
+(ns status-im.contexts.chat.group-create.events
+  (:require [legacy.status-im.data-store.chats :as data-store.chats]
+            [oops.core :as oops]
+            [re-frame.core :as rf]
+            [status-im.common.avatar-picture-picker.view :as avatar-picture-picker]))
+
+(rf/reg-event-fx :group-chat/create
+ (fn [{:keys [db]} [group-name color image]]
+   (let [selected-contacts (:group/selected-contacts db)]
+     {:json-rpc/call [{:method      "wakuext_createGroupChatWithMembers"
+                       :params      [nil group-name (into [] selected-contacts)]
+                       :js-response true
+                       :on-success  (fn [response]
+                                      (let [chat-id (-> (oops/oget response :chats)
+                                                        first
+                                                        (oops/oget :id))]
+                                        (rf/dispatch [:chat-updated response])
+                                        (rf/dispatch [:group-chat/edit
+                                                      {:chat-id    chat-id
+                                                       :group-name group-name
+                                                       :color      color
+                                                       :image      image}])))}]})))
+
+(rf/reg-event-fx :group-chat/edit-success
+ (fn [{:keys [db]} [{:keys [chat-id name color image]}]]
+   (let [new-chat {:name name :color color :image image}]
+     {:db (update-in db [:chats chat-id] #(merge % new-chat))})))
```

Comment:
interesting, why are we merging here?

Reply:
because we want to update fields in the existing chat?

Reply:
oh yes, I mean, why not just `assoc`?

Reply:
mhm :)